mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -139,17 +141,56 @@ def _get_run_artifact_root(self, run_id):
         return run_response.run.info.artifact_uri
 
-    def _get_write_credentials(self, run_id, path=None):
-        json_body = message_to_json(GetCredentialsForWrite(run_id=run_id, path=path))
-        return self._call_endpoint(
-            DatabricksMlflowArtifactsService, GetCredentialsForWrite, json_body
-        )
+    def _get_credential_infos(self, request_message_class, run_id, paths):
+        """
+        Issue one or more requests for artifact credentials, providing read or write
+        access to the specified run-relative artifact `paths` within the MLflow Run specified
+        by `run_id`. The type of access credentials, read or write, is specified by
+        `request_message_class`.
 
-    def _get_read_credentials(self, run_id, path=None):
-        json_body = message_to_json(GetCredentialsForRead(run_id=run_id, path=path))
-        return self._call_endpoint(
-            DatabricksMlflowArtifactsService, GetCredentialsForRead, json_body
-        )
+        :return: A list of `ArtifactCredentialInfo` objects providing read access to the specified
+                 run-relative artifact `paths` within the MLflow Run specified by `run_id`.
+        """
+        credential_infos = []
+        page_token = None
+
+        for paths_chunk in chunk_list(paths, _MAX_CREDENTIALS_REQUEST_SIZE):
+            while True:
+                if page_token:
+                    json_body = message_to_json(
+                        request_message_class(
+                            run_id=run_id, path=paths_chunk, page_token=page_token
+                        )
+                    )
+                else:
+                    json_body = message_to_json(
+                        request_message_class(run_id=run_id, path=paths_chunk)
+                    )
+
+                response = self._call_endpoint(
+                    DatabricksMlflowArtifactsService, request_message_class, json_body
+                )
+                credential_infos += response.credential_infos
+                page_token = response.next_page_token
+
+                if not page_token or len(response.credential_infos) == 0:
+                    break
+
+        return credential_infos
+
+    def _get_write_credential_infos(self, run_id, paths):
+        """
+        :return: A list of `ArtifactCredentialInfo` objects providing write access to the specified
+                 run-relative artifact `paths` within the MLflow Run specified by `run_id`.
+        """
+        return self._get_credential_infos(GetCredentialsForWrite, run_id, paths)
+
+    def _get_read_credential_infos(self, run_id, paths):
+        """
+        :return: A list of `ArtifactCredentialInfo` objects providing read access to the specified
+                 run-relative artifact `paths` within the MLflow Run specified by `run_id`.
+        """
+        return self._get_credential_infos(GetCredentialsForRead, run_id, paths)
```

Comment:
These methods are introduced / modified in order to leverage new batch request capabilities that have been added to `get-credentials-for-read` and `get-credentials-for-write`. Whereas each API call previously requested credentials for a single `path` within the specified `run_id`, each API call now requests credentials for up to 2000 paths in a single request.